packages/itwinui-react/src/core/utils/hooks/useResizeObserver.tsx

```diff
@@ -30,11 +30,26 @@ export const useResizeObserver = <T extends HTMLElement>(
 
       resizeObserver.current?.disconnect?.();
       if (element) {
-        resizeObserver.current = new ResizeObserver(([{ contentRect }]) =>
-          onResize(contentRect),
-        );
+        resizeObserver.current = new ResizeObserver((entries) => {
+          // We wrap onResize with requestAnimationFrame to avoid this error - ResizeObserver loop limit exceeded
+          // See: https://github.com/iTwin/iTwinUI/issues/1317
+          // See: https://stackoverflow.com/a/58701523/11547064
+          window.requestAnimationFrame(() => {
+            if (!Array.isArray(entries) || !entries.length) {
+              return;
+            }
+
+            const [{ contentRect }] = entries;
+            return onResize(contentRect);
+          });
+        });
         resizeObserver.current?.observe?.(element);
       }
+
+      return () => {
```

Comment:
L31 can be removed?

Reply:
I feel it's safer to disconnect the previous `ResizeObserver` before reassigning `resizeObserver` with a new `ResizeObserver`.

So since we're disconnecting the resize observer in both the if and the else branches, I moved it before the if-else statement. (https://github.com/iTwin/iTwinUI/pull/1800/commits/48d3be99a7d7c16ad1e891cbe06547fc844f7f59)

Reply:
How is it safer?

On mount: `resizeObserver.current` is undefined
On unmount: `element` is undefined, so `resizeObserver.current?.disconnect?.()` can safely run inside `else` block.
On remount (e.g. when the callback reference changes as a result of deps changing or something): the whole callback is invoked twice, first with `undefined` then with the correct `element`.

Reply:
Then why even introduce new cleanup code in this PR? The disconnect was already working fine, so remove the `else` branch altogether.

Reply:
Yes, the cleanup code I added was just for extra safety. But it was not necessary. Removed (https://github.com/iTwin/iTwinUI/pull/1800/commits/b0bd9c7654497c8d7f1a717c890d6b72021f6144)